miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditRunController.java

```diff
@@ -190,9 +204,13 @@ public ModelAndView setupForm(Run run, PlatformType platformType, ModelMap model
       if (run.getId() == Run.UNSAVED_ID) {
         model.put("title", "New Run");
         model.put("multiplexed", false);
+        model.put("metrics", "[]");
       } else {
         model.put("title", "Run " + run.getId());
         model.put("multiplexed", isMultiplexed(run));
+        model.put("metrics",
+            getSources().map(source -> source.fetchMetrics(run)).filter(metrics -> !LimsUtils.isStringBlankOrNull(metrics))
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor the code so this stream pipeline is used. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3958)